pkg/neorpc/result/peers.go

```diff
@@ -63,3 +64,38 @@ func (p *Peers) addPeers(addrs []string) {
 		*p = append(*p, peer)
 	}
 }
+
+func (p *Peer) UnmarshalJSON(data []byte) error {
+	type NewPeer Peer
+	var np NewPeer
+
+	err := json.Unmarshal(data, &np)
+	if err == nil {
+		*p = Peer{
+			Address: np.Address,
+			Port:    np.Port,
+		}
+		return nil
+	}
+
+	type OldPeer struct {
+		Address string `json:"address"`
+		Port    string `json:"port"`
+	}
+	var op OldPeer
+
+	err = json.Unmarshal(data, &op)
+	if err == nil {
+		port, err := strconv.Atoi(op.Port)
+		if err != nil {
+			port = 0
```

Comment:
Better return an error here.

Reply:
What is the error you want me to return here?
https://github.com/nspcc-dev/neo-go/blob/9f9c6f5dc0e9e5e8198353ca65b1fdf577bc7ed7/pkg/services/rpcsrv/server.go#L750-L755

InternalServerErrorCode ?

Reply:
This is unmarshaling code, so it's a local error (RPC works fine, but the result we have on the client is garbage), it can just `return err`.

Reply:
Ah, I incorrectly assumed you also wanted to change the `port = 0` to and error in `addPeers()`

Reply:
Those are different, server side should never have this problem.